elixir/apps/domain/priv/repo/migrations/20241216183520_migrate_email_data_again.exs

```diff
@@ -0,0 +1,22 @@
+defmodule Domain.Repo.Migrations.MigrateEmailDataAgain do
+  use Ecto.Migration
+
+  def change do
+    execute("""
+      UPDATE auth_identities AS ai
+      SET email = COALESCE(
+        CASE
+          WHEN p.adapter = 'email' OR p.adapter = 'userpass' THEN ai.provider_identifier
+          ELSE COALESCE(
+            provider_state #>> '{claims,email}',
+            provider_state #>> '{userinfo,email}'
+          )
+        END
+      )
```

Comment:
Is the outer COALESCE needed?

```suggestion
      SET email = 
        CASE
          WHEN p.adapter = 'email' OR p.adapter = 'userpass' THEN ai.provider_identifier
          ELSE COALESCE(
            provider_state #>> '{claims,email}',
            provider_state #>> '{userinfo,email}'
          )
        END
```

Reply:
Ah, good catch, I don't think it is.  I just copy/pasted that from the first one, but I'm not sure how that outer COALESCE got in the first one to begin with.  🤷‍♂️ 

Should be updated now.